pkg/proxy/iptables/proxier.go

```diff
@@ -986,8 +986,25 @@ func (proxier *Proxier) syncProxyRules() {
 					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkDropChain))...)
 				}
 			}
+		} else {
+			// REJECT rules if there is no service endpoint
+			for _, ingress := range svcInfo.LoadBalancerStatus.Ingress {
+				if ingress.IP != "" {
+					// Add a reject rule so the behavior is the same for clients going via the load balancer
+					// (immediate reject instead of connection timeout). This also avoids a stale entry in the
+					// contrack table.
+					writeLine(proxier.filterRules,
+						"-A", string(kubeForwardChain),
+						"-m", "comment", "--comment", fmt.Sprintf(`"%s loadbalancer IP has no endpoints"`, svcNameString),
+						"-m", protocol, "-p", protocol,
+						"-d", utilproxy.ToCIDR(net.ParseIP(ingress.IP)),
+						"--dport", strconv.Itoa(svcInfo.Port),
+						"-j", "REJECT",
+						"--reject-with",  "icmp-port-unreachable",
```

Comment:
none of the other REJECT use this - is it significant?

Reply:
This is the default anyway and therefore redundant. I will remove it.